paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -842,8 +851,23 @@ public final void createPartition(
     @Override
     public final void dropPartition(
             ObjectPath tablePath, CatalogPartitionSpec partitionSpec, boolean ignoreIfNotExists)
-            throws CatalogException {
-        throw new UnsupportedOperationException();
+            throws PartitionNotExistException, CatalogException {
+
+        if (!partitionExists(tablePath, partitionSpec)) {
+            throw new PartitionNotExistException(getName(), tablePath, partitionSpec);
+        }
+
+        try {
+            Identifier identifier = toIdentifier(tablePath);
+            Table table = catalog.getTable(identifier);
+            AbstractFileStoreTable fileStoreTable = (AbstractFileStoreTable) table;
+            FileStoreCommit commit = fileStoreTable.store().newCommit(UUID.randomUUID().toString());
+            List<Map<String, String>> partitions =
+                    Collections.singletonList(partitionSpec.getPartitionSpec());
+            commit.dropPartitions(partitions, BatchWriteBuilder.COMMIT_IDENTIFIER);
+        } catch (Catalog.TableNotExistException e) {
+            throw new CatalogException(e);
+        }
```

Comment:
I understand that paimon catalog should be used to drop partition. For example, when using paimon hive catalog, hive metastore needs to be updated at the same time.

Reply:
when we create a paimon partitioned table with hive catalog , [we do not save the partitions info in the hive catalog](https://github.com/apache/incubator-paimon/blob/fa68a1e91ee77aeef0c0b99160cc093e56d2405a/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java#L455) , so when we drop table partition, we only need to modify the paimon metadata.

Reply:
It's handled here.


https://github.com/apache/incubator-paimon/blob/fa68a1e91ee77aeef0c0b99160cc093e56d2405a/paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java#L479-L496

Reply:
yes， you are right, I ignored this parameter `METASTORE_PARTITIONED_TABLE`, I update it